csrc/device_lower/pass/index.cpp

```diff
@@ -1742,10 +1743,240 @@ static MmaInputSmemSwizzle getSwizzleMode(TensorView* tv) {
       swizzle->inX()->extent()->evaluate().as<int64_t>() * 16);
 }
 
+// Get the ValGroup of the ID in consumer's loop domain that corresponds to the
+// innermost dimension in the allocation domain of tv. This ID must be
+// parallelized on Mma.
+ValGroup getInnerMmaLoopGroup(TensorView* tv, const MmaOp* mma) {
+  ValGraph& id_graph = GpuLower::current()->tensorIndexer().traversalGraph();
+  auto alloc_domain = id_graph.toGroups(
+      TensorDomain::noBroadcasts(tv->getMaybeAllocationDomain()));
+  auto loop_domain =
+      id_graph.toGroups(mma->out()->as<TensorView>()->getLoopDomain());
+
+  // Start from the innermost dim in the allocation domain, propagate all the
+  // way to the consumer's loop domain, and keep track of the inner dimension.
+  // After propagating, the inner dimension should be a dimension that is
+  // parallelized on Mma.
+  ValGroup inner = nullptr;
+  for (auto it = alloc_domain.rbegin(); it != alloc_domain.rend(); ++it) {
+    if (!(*it)->front()->as<IterDomain>()->isBroadcast()) {
+      inner = *it;
+      break;
+    }
+  }
```

Comment:
At line 1752 you have already filtered out broadcasts. That doesn't mean there are no broadcasts in the ValGroup, but it does mean that this allocation domain ID is not Broadcast. So why do we need to skip ValGroups here whose first ID is broadcast?

Reply:
Good catch. I wasn't filtering out broadcasts originally in 1752, so this was needed. I changed 1752 later, but forget to clean this up.

Reply:
Removed